parser_test.go

```diff
@@ -3243,6 +3243,48 @@ func (s *testParserSuite) TestErrorMsg(c *C) {
 
 	_, _, err = parser.Parse("select 1 collate some_unknown_collation", "", "")
 	c.Assert(err.Error(), Equals, "[ddl:1273]Unknown collation: 'some_unknown_collation'")
+
+	_, _, err = parser.Parse("select std(all c1) from t", "", "")
```

Comment:
if you're truly aiming for MySQL 8.0 compatibility then `all` *must* still be supported.

```
mysql> select version();
+-----------+
| version() |
+-----------+
| 8.0.20    |
+-----------+
1 row in set (0.00 sec)

mysql> select std(all a) from a;
+------------+
| std(all a) |
+------------+
|        0.5 |
+------------+
1 row in set (0.00 sec)

mysql> select std(distinct a) from a;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'distinct a) from a' at line 1
```